### ISSUE TITLE
Notification deep linking

### ISSUE THREAD

**Issue by DonnyRozendal** [Notificare/notificare-sdk-android#53]
Hello, I'm having some trouble with retrieving a deep link URI after the app has been opened when tapping on a notification. In the dashboard, I have created a template that has a deep link rich push template. Then I send that notification to a test device. The notification does open the app. However, `intent.data` is empty. For some reason, Notificare doesn't seem to put the deep link uri into `intent.data`. Am I doing something wrong?

The deep link is retrievable in this way:
```
if (intent.action == Notificare.INTENT_ACTION_REMOTE_MESSAGE_OPENED) {
    val notification = intent.getParcelableExtra<NotificareNotification>(Notificare.INTENT_EXTRA_NOTIFICATION)
    val uri = notification?.content?.firstOrNull()?.data
}
```
However, it should also be inside `intent.data` right?

Version: 3.0.0

Setup in app delegate:
```
Notificare.launch()
Notificare.push().enableRemoteNotifications()
```
Setup in manifest (tags inside MainActivity):
```
<intent-filter>
    <action android:name="android.intent.action.VIEW" />
    <category android:name="android.intent.category.DEFAULT" />
    <category android:name="android.intent.category.BROWSABLE" />
    <data
        android:host="api-test.dpdgapp.dev"
        android:pathPrefix="/new_medal"
        android:scheme="https" />
</intent-filter>

<intent-filter>
    <action android:name="re.notifica.intent.action.RemoteMessageOpened" />
    <category android:name="android.intent.category.DEFAULT" />
</intent-filter>
```

**Comment by DonnyRozendal** [comment]
Right now, the deep link intent filter isn't utilized at all for some reason. I'm not parsing a deep link, but I'm parsing a `RemoteMessageOpened` intent. This unfortunately limits me in a way where I can't use the Android deep linking APIs.

**Comment by DonnyRozendal** [comment]
Thank you for the quick reply @hpinhal. I managed to figure it out with your help! I indeed needed to use `NOTIFICATION_OPENED` with `presentNotification`. Calling that function felt counterintuitive, since I'm not using any UI popup to present my notification. But it seems that it's necessary to still call it to correctly trigger my deep link intent filter.

For any one else who runs into this issue, some more clarification of what's actually happening:
- Clicking the notification launches the `RemoteMessageOpened` intent. So that needs to be in your Manifest.
- This intent is then used in `handleTrampolineIntent()` to correctly parse the contents, which then launches the `NotificationOpened` intent. So that also needs to be in your Manifest.
- This intent is then used to get the `NotificareNotification` by using: `intent.getParcelableExtra(Notificare.INTENT_EXTRA_NOTIFICATION)`
- This `NotificareNotification` class is then used in: `Notificare.pushUI().presentNotification(activity, notification)`
- This then launches another intent that actually triggers the deep link intent filter. Then `intent.data` actually contains the data of the deep link.

It does feel a bit awkward that the notification has to go through 3 intents before the actual deep link logic is triggered. But this does match the documentation and the sample project.
